operator/src/lib.rs

```diff
@@ -91,6 +95,50 @@ fn count_pods(hashed: &HashMap<String, Vec<Pod>>) -> usize {
 }
 
 impl SparkState {
+    /// Process the cluster status:
+    /// If no status is set, write image, version and timestamp to status
+    /// Check for restart, start, stop commands
+    pub async fn process_cluster_status(&mut self) -> SparkReconcileResult {
+        if let Some(status) = &self.status {
+            if status.version != self.spec.version {
```

Comment:
What I started doing is to keep a "current_version" and "target_version" field so that I can finish an upgrade before changing it again midway through.

I'm not sure if this is better to be honest.
But you set status.version to the spec version even if it hasn't updated yet.

Reply:
I start in a rolling fashion to ensure that masters are ready when the worker try to connect. For version updates, rolling fashion ist not really applicable (in my opinion, i did not test it so far) to the cluster for major version changes (e.g. 2.0 to 3.0). It will result in a lot of research which versions are inter compatible etc. I would like to avoid that. Since spark basically lives in its own bubble (e.g. compared to Zookeeper, where other tools depend on it) i think we do not need rolling updates? I agree though that we should stick to one way (rolling or all). I will adapt in the rolling direction for now.  And yeah i basically set the version before everything is done, that needs improvements as well. I will use your current/target approach as well to keep common features similar.